### DISCUSSION TITLE
change primary key

### DISCUSSION THREAD

**Discussion by SilvaRaulEnrique** [doctrine/doctrine2#10341] [Support Questions] [answered]
I want my primary key is not "id", want that to be "idProduct". This chage is possible?
I try
...
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $idProduct = null;
...

but a have a error:
The referenced column name 'idProduct' has to be a primary key column on the target entity class 'App\Entity\Product

It's possible change "id" to "idProduct"?

what another change i to must make?

**Comment by mpdude** [comment, marked as answer]
Nothing obviously wrong. Did you solve that?

**Comment by SilvaRaulEnrique** [comment]
If remember well, the practical solution is to create another key primary, usage 2 keys was posible but not easy, more fast solution is usage 1 only primary key in each table
